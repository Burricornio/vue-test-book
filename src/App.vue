<template>
  <div id="app">
    <ListaMensajes :mensajes="mensajes">
      <header slot="header">Header externo</header>
      <Mensaje
        v-for="(mensaje, indice) in mensajes"
        :key="indice"
        :mensaje= "mensaje"
        @mensaje-clickado="gestionarMensajeClick"
        >
      </Mensaje>
    </ListaMensajes>
    <Formulario :reversed="true" />
  </div>
</template>

<script>
import ListaMensajes from './components/ListaMensajes.vue'
import Formulario from './components/Formulario.vue'
import Mensaje from './components/Mensaje.vue'
// import axios from 'axios'

export default {
  name: 'app',
  components: {
    ListaMensajes,
    Formulario,
    Mensaje
  },
  // Forma Cesar
  // provide: {
  //   http: axios,
  // },
  data () {
    return {
      mensajes: ['Hola Don Pepito', 'Hola Don Jos√©']
    }
  },
  methods: {
    gestionarMensajeClick (mensaje) {
      console.log(mensaje)
    }
  }
}
</script>
