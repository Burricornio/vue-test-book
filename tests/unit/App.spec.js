import App from '@/App.vue'
import { shallowMount } from '@vue/test-utils'

describe('App.vue', () => {
  it('Test de comprobaciÃ³n', () => {
    const test = true
    expect(test).toBe(true)
  })

  it('Mensajes es igual a ["Item test"]', () => {
    const wrapper = shallowMount(App, {
      data () {
        return {
          mensajes: ['Item test']
        }
      }
    })
    expect(wrapper.vm.$data.mensajes).toEqual(["Item test"])
  })
})
