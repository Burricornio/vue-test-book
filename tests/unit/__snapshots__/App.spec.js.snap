// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Tiene la estructura HTML esperada 1`] = `
<div
  id="app"
>
  <listamensajes-stub
    mensajes="Hola Don Pepito,Hola Don José"
  >
    <header>
      Header externo
    </header>
     
    <mensaje-stub
      autor="Maria"
      mensaje="Hola Don Pepito"
    />
    <mensaje-stub
      autor="Maria"
      mensaje="Hola Don José"
    />
  </listamensajes-stub>
   
  <formulario-stub
    reversed="true"
  />
</div>
`;
